### ISSUE TITLE
[Bug]: ESC closes mentions dropdown, but items will still be selected and actions still forwarded to the menu

### ISSUE THREAD

**Issue by Nantris** [ueberdosis/tiptap#5662]
### Affected Packages

extension-suggestion

### Version(s)

2.7.2

### Bug Description

After pressing ESC the menu is still active, just hidden, since this PR enabled hiding it with ESC: https://github.com/ueberdosis/tiptap/pull/5544

### Browser Used

Chrome

### Expected Behavior

ESC truly closes the menu. Enter and arrow keys should affect the editor, not a hidden menu.

### Dependency Updates

- [X] Yes, I've updated all my dependencies.

**Comment by gethari** [comment]
If I understand correctly, after doing `@` when user hit `Esc` we should ideally get rid of the inserted `@` as well ? I guess that would solve this problem ?

**Comment by nperez0111** [comment]
> If I understand correctly, after doing `@` when user hit `Esc` we should ideally get rid of the inserted `@` as well ? I guess that would solve this problem ?

No, we do need to let the user to type an @ character. I think it just needs to destroy the menu until the next actual character. If the user goes next to the @ again and there is only white space to the right of it, it should reopen the menu. 

I would take a PR to fix this, but honestly it is just very annoying behavior to implement and get right

**Comment by gethari** [comment]
@nperez0111 , I am able to overcome this issue by making the below changes, let me know if I can create a PR for this

### Diff
![image](https://github.com/user-attachments/assets/41671b47-ec64-4069-9bf4-ebf7a2a19298)

### Video
https://github.com/user-attachments/assets/a732ef79-f938-46b4-b11f-66c722e98577

**Comment by rfgamaral** [comment]
@gethari Not a maintainer, but so that you know, what is only a workaround for the `Mention` extension (one that doesn't really destroy the menu), it doesn't really fix the core issue. The core issue needs to be fixed in the `Suggestion` utility, most likely.

**Comment by gethari** [comment]
@rfgamaral yeah I understand, ideally the `renderer?.onKeyDown?.` should be stopped from executing that is based on the flags getting computed above
